Tools/ardupilotwaf/git_submodule.py

```diff
@@ -111,8 +112,28 @@ def u(s):
     def __str__(self):
         return 'Submodule update: %s' % self.submodule
 
+def check_folder(path_list=[]):
+    for d in Utils.to_list(path_list):
+        listdir = os.listdir(d)
+        for f in listdir(d):
+            if f.startsith('.'):
+                listdir.remove()
+        if len(f) == 0:
+            raise
+    return f
+
 def configure(cfg):
     cfg.find_program('git')
+    path = cfg.srcnode.abspath()+'/modules/'
+    check_modules = ['ChibiOS', 'gbenchmark', 'gtest', 'libcanard', 'mavlink', 'uavcan']
+    path_list = [(path+s) for s in check_modules]
+    try: 
+        check_folder(path_list)
+    except: 
+        try:
+            subprocess.check_call(['git', 'submodule', 'update', '--init', '--recursive'])
```

Comment:
This should initialise and update  *just* the ones that don't exist.

As a convenience "you really need this module to do anything, so let's grab it" thing I'm OK with this feature, but generally playing games with other modules isn't good IMO.